hw3/task4/hw4.py

```diff
@@ -0,0 +1,25 @@
+"""
+Armstrong number is a number that is the sum of its own digits each raised to the power of the number of digits.
+https://en.wikipedia.org/wiki/Narcissistic_number
+
+Examples:
+
+- 9 is an Armstrong number, 9 = 9^1 = 9
+- 10 is not: 10 != 1^2 + 0^2 = 1
+- 153 is : 153 = 1^3 + 5^3 + 3^3 = 1 + 125 + 27 = 153
+
+
+Write a function that detects if a number is Armstrong number in functionaly style:
+ - use map or other utilities from functools library,
+ - use anonymous functions (or use function as argument)
+ - do not use loops, preferably using list comprehensions
+
+### Example function signature and call
+"""
+
+
+def is_armstrong(number: int) -> bool:
+    chars_list = str(number)
+    digits_len = len(chars_list)
+    digits_list = list(map(lambda s: int(s) ** digits_len, chars_list))
+    return sum(digits_list) == number
```

Comment:
Like adding line breaks before asserts, it is also often may be a good style to add lb before return statements, as it improves the code readability

Reply:
done